src/Npgsql/NpgsqlReadBuffer.Stream.cs

```diff
@@ -120,36 +120,51 @@ public override Task<int> ReadAsync(byte[] buffer, int offset, int count, Cancel
             }
 
             ValueTask<int> Read(byte[] buffer, int offset, int count, CancellationToken cancellationToken, bool async)
+                => Read(new Memory<byte>(buffer, offset, count), cancellationToken, async);
```

Comment:
As above, Read should call into Read(Span), ReadAsync into Read(Memory)

Reply:
Done.